src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,87 @@
-export const Pagination = () => {};
+import { FC } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+export type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (value: number) => void;
+};
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const amountOfPages = getNumbers(1, Math.ceil(total / perPage));
+
+  const handlePageChange = (event: React.MouseEvent<HTMLElement>) => {
+    const element = event.target as HTMLElement;
+    const page = element.textContent;
+
+    onPageChange(Number(page));
+  };
+
+  const handleArrowPageChange = (direction: 'prev' | 'next') => {
+    if (direction === 'next' && currentPage !== amountOfPages.length) {
+      onPageChange(currentPage + 1);
+    }
+
+    if (direction === 'prev' && currentPage !== 1) {
```

Comment:
The condition `currentPage !== 1` is correct, but ensure that the logic for decrementing the page is consistent with the logic for incrementing it.